stan-dev/stan
# Feature/issue 2467 vectorized signatures rn gs

#### Submission Checklist


#### Summary

 Vectorize the *_rng distribution functions following stan-dev/math#605 stan-dev/math#722.

#### Intended Effect

Vectorize the following functions

```
beta_rng
cauchy_rng
chi_square_rng
double_exponential_rng
exp_mod_normal_rng
exponential_rng
frechet_rng
gamma_rng
gumbel_rng
inv_chi_square_rng
inv_gamma_rng
logistic_rng
lognormal_rng
pareto_rng
pareto_type_2_rng
rayleigh_rng
scaled_inv_chi_square_rng
skew_normal_rng
student_t_rng
weibull_rng
```

#### How to Verify

Double check updated tests in  `stan/src/test/test-models/good/function-signatures/distributions/rngs.stan` and/or run

`./runTests.py src/test/unit/lang/parser/distribution_functions_test.cpp`

I didn't run `src/test/unit`, but the tests (and updated tests) from `src/test/unit/lang` do pass.

#### Side Effects

None

#### Documentation

All *_rng distribution functions have s added to their arguments to indicate these function willingly accept vectors.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

California State University, Chico

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)
